templates/eks-root-roles.yml

```diff
@@ -0,0 +1,237 @@
+              - Effect: Allow
+                Action:
+                  - s3:PutObject
+                  - s3:GetObject
+                  - s3:GetObjectVersion
+                  - s3:GetBucketAcl
+                  - s3:GetBucketLocation
+                Resource:
+                  - !Sub 'arn:${AWS::Partition}:s3:::codepipeline-${AWS::Region}-*'
+              - Effect: Allow
+                Action:
+                  - codebuild:CreateReportGroup
+                  - codebuild:CreateReport
+                  - codebuild:UpdateReport
+                  - codebuild:BatchPutTestCases
+                  - codebuild:BatchPutCodeCoverages
+                Resource:
+                  - !Sub 'arn:${AWS::Partition}:codebuild:${AWS::Region}:${AWS::AccountId}:report-group/${CodeBuildProjectName}-*'
+        - PolicyName: codebuild-s3-readonly-policy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action: 
+                  - s3:GetObject
+                  - s3:GetObjectVersion
+                Resource:
+                  - !Sub 'arn:${AWS::Partition}:s3:::${StagingS3Bucket}/buildspec.yml'
+                  - !Sub 'arn:${AWS::Partition}:s3:::${StagingS3Bucket}/*'
+              - Effect: Allow
+                Action:
+                  - s3:ListBucket
+                  - s3:GetBucketAcl
+                  - s3:GetBucketLocation
+                Resource:
+                  - !Sub 'arn:${AWS::Partition}:s3:::${StagingS3Bucket}'
+        - PolicyName: codebuild-eks-policy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - eks:DescribeFargateProfile
+                  - eks:ListAccessEntries
+                  - eks:ListAssociatedAccessPolicies
+                  - eks:DescribeNodegroup
+                  - eks:ListIdentityProviderConfigs
+                  - eks:ListNodegroups
+                  - eks:DescribeAccessEntry
+                  - eks:ListFargateProfiles
+                  - eks:DescribeIdentityProviderConfig
+                  - eks:DescribeUpdate
+                  - eks:AccessKubernetesApi
+                  - eks:DescribeCluster
+                  - eks:ListClusters
+                  - eks:ListAccessPolicies
+                  - eks:AssociateIdentityProviderConfig
+                  - eks:AssociateAccessPolicy
+                  - eks:DisassociateAccessPolicy
+                  - ecr:GetAuthorizationToken
+                  - ecr:ListImages
+                  - ecr:InitiateLayerUpload
+                  - ecr:CreateRepository
+                  - ecr:CompleteLayerUpload
+                  - ecr:PutImage
+                  - iam:CreateOpenIDConnectProvider
+                  - iam:GetOpenIDConnectProvider
+                  - ec2:CreateNetworkInterface
```

Comment:
Consider scoping down the policy by separating the CUD APIs and confine to required resources instead of `Resource: *`